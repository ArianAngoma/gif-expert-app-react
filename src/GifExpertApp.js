import {useState} from "react";
import AddCategory from "./components/AddCategory";

const GifExpertApp = () => {
    // const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
    const [categories, setCategories] = useState(['One Punch', 'Samurai X', 'Dragon Ball']);

    // Agregar elemento al array con useState
    /*const handleApp = () => {
        // setCategories([...categories, 'Anime']);
        setCategories(cats => [...cats, 'Anime']);
    }*/

    return (
        <div>
            <h2>GifExpertApp</h2>
            <AddCategory/>

            <hr/>

            <ol>
                {categories.map(category => {
                    return <li key={category}>{category}</li>
                })}
            </ol>
        </div>
    )
}

export default GifExpertApp;