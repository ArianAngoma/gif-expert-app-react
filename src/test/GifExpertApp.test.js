import {shallow} from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('Test to component <GifExpertApp/>', () => {
    let wrapper = shallow(<GifExpertApp/>);

    beforeEach(() => {
        wrapper = shallow(<GifExpertApp/>);
    })

    test('Should display the component <GifExpertApp/> correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
})