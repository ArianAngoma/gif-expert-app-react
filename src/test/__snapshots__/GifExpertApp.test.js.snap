// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to component <GifExpertApp/> Should display the component <GifExpertApp/> correctly 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</div>
`;
